tests/handlers/aws/testdata/cloudwatch-log-1.json

```diff
@@ -0,0 +1,5 @@
+{
```

Comment:
total personal preference: if the payload it's not too big I prefer to have it as variable in the test, without having to skip back and forward between two files.

Reply:
The idea behind moving that content to file is that is it possible to share some common fixtures. Doing that with variables would move them away from where they are used so I don't think is the right approach.

Reusing the same fixtures would help adding tests and normalizing input data across the test suite (I extracted three of them, I've no idea how many of there there are in the current tests)

Reply:
Indeed I would not focus on reusing the same fixtures: if we did so far it was probably because of laziness ;)

The more fuzziness we could add to the fixtures the better it would be in my opinion: given the experience of the unit tests for storage, having the content totally random, helped to catch corner case bugs we would not have spotted otherwise.

but I'm find with your current solution for the moment :)

Reply:
My understanding is that testing and fuzzing have different scope and should happen at different times. 

We need tests (unit/integration) to be reliable, predictable and easy to reproduce; that may not happen with random inputs or inputs that change through test runs.

On the other end we need fuzzing to catch errors with data that we cannot foresee or that we have not thought about.

Instead of trying to do both with our current fixtures, what about implementing fuzz test as separate tests?

Reply:
> Instead of trying to do both with our current fixtures, what about implementing fuzz test as separate tests?

if we do both, separated, I'm oki, but we should try to add them together, or at least have both by the time a feature is released.

if we do only one of that, I would prefer to have fuzzing over easy to reproduce tests with fixed fixtures: while the first are harder to maintain, they give broader coverage to corner cases and I prefer the trade off of an higher effort on the development side with the benefit of catching as much bugs as possible.

As long as I can understand clearly what test with what fuzziness init parameters is failing, and I have a way to replay the same random content.

The last is not 100% true at the moment: when a storage unit test failed because of the random content case we didn't manage properly, I had to add some code in a local branch to dump the random content to a file, and then run the test again until the same content case will occur. After that I have the content that gives a reliable and predictable test.

Yes, it's not easy to reproduce: 
> I had to add some code in a local branch to dump the random content to a file

this should be done by default, without needing to add code back and forth, and I should have at the same time the way to run a test telling if loading a dumped content or not. those artifacts should be stored by the CI job and I should be able to download them easily and feed them easily to the test.

if we have the above, and all this information is easy to access and be replayed, and it's easy to spot the failing test etc, and the test is not flaky by itself and there is a real bug in the tested code, I personally consider it "to be reliable, predictable and easy to reproduce" :)

The effort we should avoid is about understanding if a test run or not, if it failed, with what parameters, if we can trust the test in the first place, etc etc ;)